src/emulator/auth/operations.ts

```diff
@@ -1460,7 +1460,43 @@ function signInWithIdp(
     }
   }
 
-  let { response, rawId } = fakeFetchUserInfoFromIdp(providerId, claims);
+  if (reqBody.pendingToken) {
+    const decoded = decodeJwt(reqBody.pendingToken, { complete: true }) as { expiresAt: number };
+    assert(decoded.expiresAt < Date.now(), "INVALID_PENDING_TOKEN");
+  }
+
+  const encryptedSessionState =
```

Comment:
Similarly, I don't think Auth Emulator generates `state` in `createAuthUri` -- using that method for IDP sign-in is not yet supported: https://github.com/firebase/firebase-tools/blob/466036dd4a8f3005b3c1d6a132c2b5dd8f12e9ed/src/emulator/auth/operations.ts#L557